astropy/io/fits/hdu/compressed.py

```diff
@@ -676,7 +676,7 @@ def match_header(cls, header):
         if xtension not in ('BINTABLE', 'A3DTABLE'):
             return False
 
-        if 'ZIMAGE' not in header or header['ZIMAGE'] != True:
+        if 'ZIMAGE' not in header or header['ZIMAGE'] is not True:
```

Comment:
I would remove `!= True` (as you did above), i.e., let `header['ZIMAGE']` define truthiness itself.